snapcraft/internal/elf.py

```diff
@@ -301,14 +306,17 @@ def __init__(self, *, dynamic_linker: str, root_path: str) -> None:
         # in by packaging dependencies.
         # The docker conditional will work if the docker image has the
         # snaps unpacked in the corresponding locations.
-        if common.is_snap():
+        which_patchelf = shell_utils.which('patchelf')
+        if which_patchelf.endswith('stage/patchelf'):
```

Comment:
Isn't this flaky? It could point to somewhere else completely different. I would say that this needs to check the full stage path.